### ISSUE TITLE
Components: warning message for double id's

### ISSUE THREAD

**Issue by verpeteren** [nidium/Nidium#80]
I am experimenting with the components branch (6418e6c) 
And found a minor thing that needs a little clarificaiton:

The following nml has 2 buttons with the same 'id'.

```xml
<application>

<meta>
        <viewport>100x100</viewport> 
</meta> 

<layout>
        <button id="1" class="here">1</button>
        <button id="1" class="here">2</button>
</layout>

<script>
        document.getElementById("1").style.backgroundColor = "red";
</script>

</application>
```

This gives the following message:

```
[WARNING] shadowRoot already has an element with id 1. Adding new element will override current element with same id
```

The output does still show 2 distinct buttons.

![screenshot_20170504_094837](https://cloud.githubusercontent.com/assets/4726682/25694414/df18aa04-30ae-11e7-8410-c0a702d6ca64.png)

The output is not consistent with the warning; I would expect:

* warning as it is now, but with only 1 button
* warning as it is now, but with 2 identical buttons
* output as it is now, but with a different text ('... The element that had this Id, will loose it)
* ..

What do you think?

**Comment by greduan** [comment]
I think this behaviour is similar in other browsers.

Using `id` actually sets a global variable in JS for that ID, so `<button id="boop" />` means that in the console `window.boop` exists and is pointing to the DOM element. I think it is normal for it to render twice properly, this is the case with other browser as well.

I think this behaviour is fine. But perhaps something more show stopping should happen, not sure.

**Comment by efyx** [comment]
Hi, I believe the behavior we have right now is correct (the two buttons must be displayed), but the message is indeed missleading. What is actually overriden is the reference to the Element (when you use `getElementById()`) not the Element itself.

What do you think about `[WARNING] shadowRoot already has an element with id "1". Previous element reference will be lost.`

**Comment by greduan** [comment]
Perhaps `Reference to previous element will be lost` is clearer

**Comment by efyx** [comment]
Thanks for your feedbacks. This issue has been fixed in commit 5efb81b1ce00eaccf7536513105ca077266dc26e.
